clients/v1_repl/v1_repl.go

```diff
@@ -90,18 +135,128 @@ func (c *Client) Create(ctx context.Context) error {
 	version := res.Header.Get("X-Influxdb-Version")
 	color.Cyan("Connected to InfluxDB %s %s", build, version)
 
-	c.Databases, _ = c.GetDatabases(ctx)
-	color.Cyan("InfluxQL Shell")
-
-	scanner := bufio.NewScanner(os.Stdin)
-	fmt.Printf("%s", color.GreenString("> "))
-	for scanner.Scan() {
-		c.executor(scanner.Text())
-		fmt.Printf("%s", color.GreenString("> "))
+	// compute historyFilePath at REPL start
+	// Only load/write history if HOME environment variable is set.
+	var historyDir string
+	if runtime.GOOS == "windows" {
+		if userDir := os.Getenv("USERPROFILE"); userDir != "" {
+			historyDir = userDir
+		}
 	}
+	if homeDir := os.Getenv("HOME"); homeDir != "" {
+		historyDir = homeDir
+	}
+	var history []string
+	if historyDir != "" {
+		c.historyFilePath = filepath.Join(historyDir, ".influx_history")
+		history = c.readHistory()
+		c.rewriteHistoryFile(history)
```

Comment:
For each command, we write to the history file, but there was no place where the file would be bounded in size. We'd have to read the entire file and pull out the last 100 entries, but it would only ever get bigger.
I figured that the start of the REPL was a good time to truncate the file back down to only the latest 100 commands.

Reply:
That's worth a comment I think.